source/agora/test/GossipProtocol.d

```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+
+    Contains tests for Gossip Protocol.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.GossipProtocol;
+
+import agora.common.Data;
+
+import agora.protocol.GossipProtocol;
+
+import std.digest.sha;  
+ 
+ /// Test API
+interface TestAPI {
+
+    public bool setMessage(Hash msg);
+}
+
+/// Remote Node for test
+class TestRemoteNode
+{
+    /// API client to the node
+    private TestAPI api;
+
+    /// Ctor
+    public this (TestAPI api)
+    {
+        this.api = api;
+    }
+
+    /// Send a message
+    public bool sendMessage(Hash msg) {
+        return this.api.setMessage(msg);
+    }
+}
+
+/// Network for test
+class TestNetwork {
+    private TestRemoteNode[] remoteNodes;
+
+    /// Ctor
+    public this () {
+    }
+
+    public void setRemotes (TestAPI [] api) {
+        for (int idx = 0; idx < api.length; idx++) {
```

Comment:
There is a simpler way to write this in D:

```D
public void setRemotes (TestAPI[] apis)
{
    foreach (api; apis)
    {
        this.remoteNodes ~= TestRemodeNode(api);
    }
}
```

Note I've renamed the `api` parameter to `apis` to make it more obvious it's an array and not a single API instance.

Reply:
Modified.

Reply:
@AndrejMitrovic, Removed this code,
Check Please~